packages/merge-styles/src/concatStyleSets.ts

```diff
@@ -1,30 +1,50 @@
+import { IStyleSet } from './IStyleSet';
+import { IStyleOrStyleFunction } from './IStyle';
+
 /**
  * Combine a set of styles together (but does not register css classes.)
  * @public
  */
-export function concatStyleSets<T extends object>(...args: (T | false | null | undefined)[]): T {
-  // tslint:disable-next-line:no-any
+export function concatStyleSets(...args: (IStyleSet<any> | false | null | undefined)[]): IStyleSet<any> {
   const mergedSet = {} as any;
 
   for (const currentSet of args) {
     if (currentSet) {
       for (const prop in currentSet) {
         if (currentSet.hasOwnProperty(prop)) {
           const mergedValue = mergedSet[prop];
-          const currentValue = currentSet[prop];
+          const currentValue: IStyleOrStyleFunction = currentSet[prop];
 
           if (mergedValue === undefined) {
             mergedSet[prop] = currentValue;
           } else {
-            mergedSet[prop] = [
-              ...(Array.isArray(mergedValue) ? mergedValue : [mergedValue]),
-              ...(Array.isArray(currentValue) ? currentValue : [currentValue])
-            ];
+            if (typeof currentValue === 'function') {
+              // if it is a function, we need special handling.
+              mergedSet[prop] = (props: any) => {
+                if (typeof mergedValue === 'function') {
+                  return concatStyleSets(mergedValue(props), currentValue(props));
```

Comment:
👍

Reply:
This works, yes, but your else could be supported just fine.

This is why I used _list in my example i showed you yesterday. Build up an array of things. If function, call function to get an object. In the end, have an array of objects and concat them.

Reply:
Our typings at present don't allow that - it is easy to support it otherwise (it feels a bit pointless if the typings don't support - you run into friction everywhere including in the unit tests and have to rely on unnatural things like cast to any to get around). We could spend the time to tweak the typings to support this scenario, but it would add a bunch more time to completing this work and I think we might as well wait until there is a real need for this scenario.